clients/admin-ui/src/features/common/custom-fields/hooks.ts

```diff
@@ -93,23 +93,35 @@ export const useCustomFields = ({
     [activeCustomFieldDefinition]
   );
 
-  const definitionIdToCustomField = useMemo(() => {
-    // @ts-ignore
-    if (isError && error?.status === 404) {
-      return new Map();
-    }
-    const newMap = new Map(
-      filterWithId(data).map((fd) => [fd.custom_field_definition_id, fd])
-    );
-    return newMap;
-  }, [data, isError, error]);
+  const definitionIdToCustomField: Map<string, CustomFieldWithId> =
+    useMemo(() => {
+      // @ts-ignore
+      if (isError && error?.status === 404) {
+        return new Map();
+      }
+      const newMap = new Map(
+        filterWithId(data).map((fd) => [fd.custom_field_definition_id, fd])
+      );
+      return newMap;
+    }, [data, isError, error]);
 
   const sortedCustomFieldDefinitionIds = useMemo(() => {
     const ids = [...idToCustomFieldDefinition.keys()];
     ids.sort();
     return ids;
   }, [idToCustomFieldDefinition]);
 
+  /**
+   * Transformed version of definitionIdToCustomField to be easy
+   * to pass into Formik
+   */
+  const customFieldValues: Record<string, string | string[]> = {};
+  if (definitionIdToCustomField) {
+    definitionIdToCustomField.forEach((value, key) => {
+      customFieldValues[key] = value.value.toString();
+    });
+  }
```

Comment:
It's possible that it's not required for things to work properly. Sometimes due to the way the tree is structured it'll cause hook functions like this one to run every time a value like `customFieldValues` needs to re-initialized. `useMemo` and `useEffect` are helpful when you want control over when some code does or doesn't run. Since it's probably cheap to re-initialize `customFieldValues` a `useMemo` may not be needed. It wouldn't hurt to put it in one though.

Does that seem correct in this situation @allisonking?

Reply:
yeah that makes sense—I'll wrap it in a `useMemo` 👍

Reply:
whoops sorry I didn't see @TheAndrewJackson 's comment before I commented 😅 

I do think your analysis is correct—`useMemo` in general is for expensive operations since there is a slight cost to using `useMemo` in the first place. as for what counts as "expensive", I'm not up to date on. but I don't think it hurts to put the `useMemo` in now, and if we're really coming down to perf issues, then to look into this hook more